omnigibson/prims/entity_prim.py

```diff
@@ -954,76 +954,138 @@ def get_relative_linear_velocity(self):
+        return T.quat2mat(self.get_position_orientation()[1]).T @ self.get_angular_velocity()
 
-    def set_position_orientation(self, position=None, orientation=None):
-        # If kinematic only, clear cache for the root link
-        if self.kinematic_only:
-            self.root_link.clear_kinematic_only_cache()
-        # If the simulation isn't running, we should set this prim's XForm (object-level) properties directly
-        if og.sim.is_stopped():
-            XFormPrim.set_position_orientation(self, position=position, orientation=orientation)
-        # Delegate to RigidPrim if we are not articulated
-        elif self._articulation_view is None:
-            self.root_link.set_position_orientation(position=position, orientation=orientation)
-        # Sim is running and articulation view exists, so use that physx API backend
-        else:
-            if position is not None:
-                position = np.asarray(position)[None, :]
-            if orientation is not None:
-                orientation = np.asarray(orientation)[None, [3, 0, 1, 2]]
-            self._articulation_view.set_world_poses(position, orientation)
-            PoseAPI.invalidate()
+    def set_position_orientation(self, position=None, orientation=None, frame=RelativeFrame.WORLD):
+        """
+        Set the position and orientation of entry prim object.
+
+        Args:
+            position (None or 3-array): The position to set the object to. If None, the position is not changed.
+            orientation (None or 4-array): The orientation to set the object to. If None, the orientation is not changed.
+            frame (RelativeFrame): The frame in which to set the position and orientation. Defaults to WORLD. PARENT frame
+            set position relative to the object parent. SCENE frame set position relative to the scene.
+        """
+
+        if frame == RelativeFrame.WORLD:
+
+            # If kinematic only, clear cache for the root link
+            if self.kinematic_only:
+                self.root_link.clear_kinematic_only_cache()
+            # If the simulation isn't running, we should set this prim's XForm (object-level) properties directly
+            if og.sim.is_stopped():
+                XFormPrim.set_position_orientation(self, position=position, orientation=orientation)
+            # Delegate to RigidPrim if we are not articulated
+            elif self._articulation_view is None:
+                self.root_link.set_position_orientation(position=position, orientation=orientation)
+            # Sim is running and articulation view exists, so use that physx API backend
+            else:
+                if position is not None:
+                    position = np.asarray(position)[None, :]
+                if orientation is not None:
+                    orientation = np.asarray(orientation)[None, [3, 0, 1, 2]]
+                self._articulation_view.set_world_poses(position, orientation)
+                PoseAPI.invalidate()
```

Comment:
Notice that almost all of the work here is identical to the one in the local. Do the branching carefully only for the articulation view setter line.